src/domain-analysis.js

```diff
@@ -0,0 +1,44 @@
+/* This file contains the code for analyzing number of blog posts from different domains.
+ * The module receives an array of blog urls and returns an array of json object in format
+ * {name: domainName, count: number of frequency}
+*/
+
+
+module.exports.countDomain = function (url) {
+    var domainCounter = []
+
+    for (let i = 0; i < url.length; i++ ){
+        let domain = findDomain(url[i])
+        let domainObject = {}
+        if(domainCounter.length == 0){
+            domainObject.name = domain;
+            domainObject.count = 1;
+            domainCounter.push(domainObject);
+        }
+        else{
+            let flag = false
+            for(let j = 0; j < domainCounter.length; j++){
+                if(domain == domainCounter[j].name){
+                    flag = true
+                    domainCounter[j].count++
+                }
+            }
+            if(flag == false){
+                domainObject.name = domain;
+                domainObject.count = 1;
+                domainCounter.push(domainObject);
+            }
+        }
+    }
+    return domainCounter;
+};
+
+function countDomain(url){
+    
+}
+
+function findDomain(url){
+    var userDomain = url.replace('http://','').replace('https://','').split('/')[0];
```

Comment:
It's really hard to do this for all cases with a URL (i.e., operate on strings).  Instead, use node's built in [`url` module and its `parse` method](https://nodejs.org/api/url.html#url_url_parse_urlstring_parsequerystring_slashesdenotehost).  Here's how it works:

```js
const url = require('url');
url.parse('https://github.com/Seneca-CDOT/telescope/pull/166/files');
/* gives you an object like this:
Url {
  protocol: 'https:',
  slashes: true,
  auth: null,
  host: 'github.com',
  port: null,
  hostname: 'github.com',
  hash: null,
  search: null,
  query: null,
  pathname: '/Seneca-CDOT/telescope/pull/166/files',
  path: '/Seneca-CDOT/telescope/pull/166/files',
  href: 'https://github.com/Seneca-CDOT/telescope/pull/166/files' 
}
*/
```